src/main/java/io/neonbee/data/DataQuery.java

```diff
@@ -54,7 +56,7 @@ public final class DataQuery { // NOPMD not a "god class"
 
     @VisibleForTesting
     @JsonProperty
-    Map<String, List<String>> headers;
+    Map<String, List<String>> headers = new CaseInsensitiveMap<>();
```

Comment:
```suggestion
    Map<String, List<String>> headers = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```
Then we would not need the dependency on commons-collections4.